lib/ansible/plugins/loader.py

```diff
@@ -296,6 +383,37 @@ def _find_plugin(self, name, mod_type='', ignore_deprecated=False, check_aliases
             # they can have any suffix
             suffix = ''
 
+        # TODO: we should still probably require a well-known prefix on FQNS so we don't paint ourselves into a corner with future stuff
+        if ('.' in name or collection_list) and not name.startswith('Ansible'):  # HACK: need this right now so we can still load shipped PS module_utils
+            if '.' in name or not collection_list:
+                candidates = [name]
+            else:
+                candidates = ['{0}.{1}'.format(c, name) for c in collection_list]
+            # TODO: keep actual errors, not just assembled messages
+            errors = []
+            for candidate_name in candidates:
+                try:
+                    # HACK: refactor this properly
+                    if candidate_name.startswith('ansible.legacy'):
+                        # just pass the raw name to the old lookup function to check in all the usual locations
+                        p = self._find_plugin_legacy(name.replace('ansible.legacy.', '', 1), ignore_deprecated, check_aliases, suffix)
+                    else:
+                        p = self._find_fq_plugin(candidate_name, suffix)
+                    if p:
+                        return p
+                except Exception as ex:
+                    errors.append(to_native(ex))
+
+            if errors:
+                raise Exception('; '.join(errors))
```

Comment:
Or maybe even better
```suggestion
                raise LookupError('; '.join(errors))
```